### PR TITLE
rework zinger as kafka sync

### PR DESCRIPTION
**Pull request by mccanne**

### CODE REVIEW

File: fifo/producer.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+				if ev.TopicPartition.Error != nil {
+					fmt.Printf("Failed to deliver message: %v\n", ev.TopicPartition)
+				} else {
+					//fmt.Printf("Successfully produced record to topic %s partition [%d] @ offset %v\n",
+					//	*ev.TopicPartition.Topic, ev.TopicPartition.Partition, ev.TopicPartition.Offset)
+				}
+			}
+		}
+	}()
+	var cnt int
```
Comment (nwt) [inline fifo/producer.go:53]: This feels like an unfortunate abbreviation. Maybe call it `count` or `n`?

File: fifo/producer.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		Key:   nil,
+		Value: b}, nil)
+	return err
+}
+
+func (p *Producer) CreateSchema(schema []byte) (int, error) {
+	// We use RecordNameStrategy for the subject name so we can have
+	// different schemas on the same topic.
+	subject := string(schema)
+	s, err := p.registry.CreateSchema(subject, string(schema), srclient.Avro)
```
Comment (nwt) [inline fifo/producer.go:113]:
```suggestion
func (p *Producer) CreateSchema(schema string) (int, error) {
	// We use RecordNameStrategy for the subject name so we can have
	// different schemas on the same topic.
	subject := schema
	s, err := p.registry.CreateSchema(subject, schema, srclient.Avro)
```

File: zavro/decoder.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	}
+	b.EndContainer()
+	return in, nil
+}
+
+func decodeArray(b *zcode.Builder, in []byte, schema *avro.ArraySchema) ([]byte, error) {
+	elemType := schema.Items
+	for {
+		// XXX check for size exceeded on array that doesn't fit in mem
+		var cnt int64
```
Comment (nwt) [inline zavro/decoder.go:56]: `count` or `n`?

File: zavro/encoder.go [hunk 4]
```diff
@@ -225,74 +227,54 @@ func encodeScalar(dst []byte, typ zng.Type, body zcode.Bytes) ([]byte, error) {
...
-		return appendVarint(dst, int64(port)), nil
-
-	case zng.IdString, zng.IdBstring:
-		s := typ.StringOf(body, zng.OutFormatZeek, false)
-		return appendCountedValue(dst, []byte(s)), nil
-
-	case zng.IdNet:
+		return appendVarint(dst, int64(us)), nil
+	case zng.IDString, zng.IDBstring:
+		//s := typ.StringOf(body, zng.OutFormatZeek, false)
```
Comment (nwt) [inline zavro/encoder.go:259]:
```suggestion
```

File: zavro/schema.go [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+		schema, err := EncodeSchema(col.Type, namespace)
+		if err != nil {
+			return nil, err
+		}
+		var union [2]avro.Schema
+		union[0] = &avro.NullSchema{}
+		union[1] = schema
+		fld := &avro.SchemaField{
+			Name: col.Name,
+			Type: &avro.UnionSchema{union[:]},
```
Comment (nwt) [inline zavro/schema.go:39]:
```suggestion
			Type: &avro.UnionSchema{
				Type: []avro.Schema{&avro.NullSchema{}, schema},
			},
```

**Review by nwt — APPROVED** [review]
:shipit: — all my comments are just nits!

File: cmd/zinger/ls/command.go [hunk 6]
```diff
@@ -1,25 +1,23 @@
...
 
 var Ls = &charm.Spec{
 	Name:  "ls",
 	Usage: "ls [-l] [schemaID ... ]",
 	Short: "list schemas",
 	Long: `
-The ls command allows you to list the schema IDs under the specified namespace
-and subject. Specify the -l argument causes the content of each schema to be
-displayed as NDJSON.
+The ls command allows prints the schema registry subjects and schemas information
```
[the remark this answers is not in the document]
Comment (mccanne) [reply cmd/zinger/ls/command.go:18]: prints is ok here

File: cmd/zinger/sync/command.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+	"flag"
+
+	"github.com/brimdata/zed/pkg/charm"
+	"github.com/brimdata/zinger/cmd/zinger/root"
+)
+
+var Sync = &charm.Spec{
+	Name:  "sync",
+	Usage: "sync [options]",
+	Short: "sync a pool in a Zed lake to a kafka topic",
```
[the remark this answers is not in the document]
Comment (mccanne) [reply cmd/zinger/sync/command.go:14]: We need to get from/to right as they both happen here.  Subsequent PR.

**Pull request merged**